backend/model_manager.py

```
@@ -414,7 +414,152 @@
+    }
+
+    models = []
+    for model in AVAILABLE_MODELS:
+        model_copy = dict(model)
+        model_copy["recommended_for_system"] = model["id"] in recommended_ids
+        model_copy["compatibility"] = compatibility.get(model["id"], "not_recommended")
+        models.append(model_copy)
+    return models
+
+
+def _coerce_number(value, default):
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `_coerce_number`

(ANN202)